sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -0,0 +1,186 @@
+package azcosmos
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	azlog "github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
+	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/log"
+)
+
+const globalEndpointManagerDefaultBackgroundRefreshInterval = 5 * time.Minute
+
+type globalEndpointManager struct {
+	client           *Client
+	preferredRegions []string
+	locationCache    *locationCache
+	refreshTicker    *time.Ticker
+	refreshDone      chan struct{}
+	sync.Mutex
+}
+
+// NewGlobalEndpointManager creates a new global endpoint manager.
+func newGlobalEndpointManager(client *Client, preferredRegions []string, refreshInterval time.Duration) (*globalEndpointManager, error) {
+	endpoint, err := url.Parse(client.endpoint)
+	if err != nil {
+		return nil, err
+	}
+	gem, err := &globalEndpointManager{
+		client:           client,
+		preferredRegions: preferredRegions,
+		locationCache:    newLocationCache(preferredRegions, *endpoint),
+	}, nil
+	if err != nil {
+		return nil, err
+	}
+	gem.stopBackgroundRefresh()
```

Comment:
What is the purpose of stopping the background refresh if it's not started?

Reply:
On new instance refresh background task is not yet started, its not necessary right?